crates/sui-json-rpc/src/read_api.rs

```diff
@@ -121,6 +125,60 @@ impl SuiRpcModule for ReadApi {
 
 #[async_trait]
 impl RpcFullNodeReadApiServer for FullNodeApi {
+    async fn get_move_function_arg_types(
+        &self,
+        package: ObjectID,
+        module: String,
+        function: String,
+    ) -> RpcResult<Vec<MoveFunctionArgType>> {
+        let object_read = self
+            .state
+            .get_object_read(&package)
+            .await
+            .map_err(|e| anyhow!("{e}"))?;
+
+        let normalized = match object_read {
+            ObjectRead::Exists(_obj_ref, object, _layout) => match object.data {
+                Data::Package(p) => normalize_modules(p.serialized_module_map().values())
+                    .map_err(|e| anyhow!("{e}")),
+                _ => Err(anyhow!("Object is not a package with ID {}", package)),
+            },
+            _ => Err(anyhow!("Package object does not exist with ID {}", package)),
+        }?;
+
+        let identifier = Identifier::new(function.as_str()).map_err(|e| anyhow!("{e}"))?;
+        let parameters = normalized.get(&module).and_then(|m| {
+            m.exposed_functions
+                .get(&identifier)
+                .map(|f| f.parameters.clone())
+        });
+
+        Ok(match parameters {
+            Some(parameters) => Ok(parameters
+                .iter()
+                .map(|p| match p {
+                    Type::Bool => MoveFunctionArgType::Bool,
+                    Type::U8 => MoveFunctionArgType::U8,
+                    Type::U64 => MoveFunctionArgType::U64,
+                    Type::U128 => MoveFunctionArgType::U128,
+                    Type::Address => MoveFunctionArgType::Address,
+                    Type::Signer => MoveFunctionArgType::Signer,
+                    Type::Struct {
+                        address: _,
+                        module: _,
+                        name: _,
+                        type_arguments: _,
+                    } => MoveFunctionArgType::Struct,
+                    Type::Vector(_) => MoveFunctionArgType::Vector,
```

Comment:
If you are going to do this, why not just provide the full type? `Vector` or `TypeParameter` or `Struct` don't seem that helpful to me unless I also know the whole type?

If you want to capture the kind/intent of the argument here, I might suggest
```
Pure,
Object(ObjectValueKind),
```
where
```
ObjectValueKind {
  ByImmutableReference,
  ByMutableReference,
  ByValue,
}
```

Reply:
agreed that here Vector is probably not very useful and the suggested type should be enough for the wallet use-case, @666lcz mentioned that sometime types like U64 can be useful. Chris can you add some contexts here?

Reply:
I think you should have:
(A) full type information
(B) pure or object+kind
(C) nothing :p 

Having some weird mismatch of partial type information I think will be confusing as it will not line up with how Move will be looking at these values